### PR TITLE
[Communication] - Remove nullable ref-types from sms package

### PR DESCRIPTION
**Pull request by nmaredia**

### CODE REVIEW

File: sdk/communication/Azure.Communication.Sms/src/SmsClient.cs [hunk 1]
```diff
@@ -34,7 +34,7 @@ public SmsClient(Uri endpoint, AzureKeyCredential keyCredential, SmsClientOption
         /// <summary> Initializes a new instance of <see cref="SmsClient"/>.</summary>
         /// <param name="connectionString">Connection string acquired from the Azure Communication Services resource.</param>
         /// <param name="options">Client option exposing <see cref="ClientOptions.Diagnostics"/>, <see cref="ClientOptions.Retry"/>, <see cref="ClientOptions.Transport"/>, etc.</param>
-        public SmsClient(string connectionString, SmsClientOptions? options = default)
+        public SmsClient(string connectionString, SmsClientOptions options = default)
```
Comment (RezaJooyandeh) [inline sdk/communication/Azure.Communication.Sms/src/SmsClient.cs:37]: let's split this constructor into two. One with only `connectionString` and one with both `connectionString` and non-optional `options` to follow the guidelines. The rest of constructors can stay as is. Only the basic constructor needs the simple format. https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-client

**Review by RezaJooyandeh — APPROVED** [review]
Thank you for quickly fixing this 🙂

**Review by RezaJooyandeh — APPROVED** [review]

**Comment by nmaredia** [comment]
azp run net - communication - ci

**Pull request closed without merging**